shim/src/main.rs

```diff
@@ -445,42 +280,70 @@ fn get_version() -> &'static str {
         .0
 }
 
+/// Checks for `TURBO_BINARY_PATH` variable. If it is set,
+/// we do not do any inference, we simply run the command as
+/// the current binary. This is due to legacy behavior of `TURBO_BINARY_PATH`
+/// that lets users dynamically set the path of the turbo binary. Because
+/// inference involves finding a local turbo installation and executing that
+/// binary, these two features are fundamentally incompatible.
+fn is_turbo_binary_path_set() -> bool {
+    env::var("TURBO_BINARY_PATH").is_ok()
+}
+
 fn main() -> Result<()> {
-    let clap_args = Args::parse();
-    // --help doesn't work with ignore_errors in clap.
-    if clap_args.help {
-        let mut command = Args::command();
-        command.print_help()?;
-        process::exit(0);
-    }
-    // --version flag doesn't work with ignore_errors in clap, so we have to handle
-    // it manually
-    if clap_args.version {
-        println!("{}", get_version());
-        process::exit(0);
-    }
+    let mut clap_args = Args::new()?;
 
     let current_dir = if let Some(cwd) = &clap_args.cwd {
         fs::canonicalize::<PathBuf>(cwd.into())?
     } else {
         env::current_dir()?
     };
 
-    let args: Vec<_> = env::args().skip(1).collect();
-    if args.is_empty() {
-        process::exit(1);
+    clap_args.cwd = Some(current_dir.to_string_lossy().to_string());
+    if clap_args.test_run {
+        println!("{:#?}", clap_args);
+    }
+    // If there is no command, we set the command to `Command::Run` with
+    // `self.parsed_args.run_args` as arguments.
+    if clap_args.command.is_none() {
+        if let Some(run_args) = take(&mut clap_args.run_args) {
```

Comment:
Super nit, `take` is a generic enough of a function name, that keeping it qualified would be helpful for readability.
```suggestion
        if let Some(run_args) = mem::take(&mut clap_args.run_args) {
```